test/syscalls/linux/dev.cc

```diff
@@ -146,6 +158,24 @@ TEST(DevTest, WriteDevFull) {
   EXPECT_THAT(WriteFd(fd.get(), "a", 1), SyscallFailsWithErrno(ENOSPC));
 }
 
+TEST(DevTest, ReadWriteDevKmsg) {
+  const FileDescriptor fd =
+      ASSERT_NO_ERRNO_AND_VALUE(Open("/dev/kmsg", O_WRONLY));
+  constexpr int overMaxSize = 2048;
+  std::vector<char> overMaxBuf(overMaxSize, 1);
+  EXPECT_THAT(WriteFd(fd.get(), overMaxBuf.data(), overMaxSize), SyscallFailsWithErrno(EINVAL));
+  constexpr int bufferSize = 512;
+  std::vector<char> buf(bufferSize, 1);
+  EXPECT_THAT(WriteFd(fd.get(), buf.data(), bufferSize),
+              SyscallSucceedsWithValue(bufferSize));
+  EXPECT_THAT(close(fd), SyscallSucceeds())
```

Comment:
`close(fd.release(), SyscallSucceeds())`. Otherwise the `FileDescriptor` retains ownership of a now-invalid fd, and attempts to close it during it's [destructor](https://cs.opensource.google/gvisor/gvisor/+/master:test/util/file_descriptor.h;l=71?q=FileDescriptor&ss=gvisor%2Fgvisor).